### ISSUE TITLE
Makefile:80: *** missing separator.  Stop.

### ISSUE THREAD

**Issue by lazarusfriday11** [rpuntaie/rstdoc#1]
I thought I had it working for a while but I keep hitting this missing separator error. I'm not well versed enough in all this to troubleshoot by fixing syntax. 

# Makefile

```
SPHINXOPTS  = -c .
SPHINXBLD   = sphinx-build
SPHINXPROJ  = sample
DOCDIR      = doc/
DOCBACK     = ../
DCXFROMDOC  = ../
BLDDIR      = build/doc/
STPLS       = $(wildcard $(DOCDIR)*.stpl)
STPLTGTS    = $(STPLS:%.stpl=%)
SRCS        = $(filter-out $(DOCDIR)index.rest,$(wildcard $(DOCDIR)*.rest))
SRCSTPL     = $(wildcard $(DOCDIR)*.rest.stpl)
IMGS        = \
	$(wildcard $(DOCDIR)*.pyg)\
	$(wildcard $(DOCDIR)*.eps)\
	$(wildcard $(DOCDIR)*.tikz)\
	$(wildcard $(DOCDIR)*.svg)\
	$(wildcard $(DOCDIR)*.uml)\
	$(wildcard $(DOCDIR)*.dot)\
	$(wildcard $(DOCDIR)*.eps.stpl)\
	$(wildcard $(DOCDIR)*.tikz.stpl)\
	$(wildcard $(DOCDIR)*.svg.stpl)\
	$(wildcard $(DOCDIR)*.uml.stpl)\
	$(wildcard $(DOCDIR)*.dot.stpl)
PNGS=$(subst $(DOCDIR),$(DOCDIR)_images/,\
	$(patsubst %.eps,%.png,\
	$(patsubst %.pyg,%.png,\
	$(patsubst %.tikz,%.png,\
	$(patsubst %.svg,%.png,\
	$(patsubst %.uml,%.png,\
	$(patsubst %.dot,%.png,\
	$(patsubst %.eps.stpl,%.png,\
	$(patsubst %.dot.stpl,%.png,\
	$(patsubst %.tikz.stpl,%.png,\
	$(patsubst %.svg.stpl,%.png,\
	$(patsubst %.uml.stpl,%.png,$(IMGS)))))))))))))
DOCXS = $(subst $(DOCDIR),$(BLDDIR)docx/,$(SRCS:%.rest=%.docx))\
	$(subst $(DOCDIR),$(BLDDIR)docx/,$(SRCSTPL:%.rest.stpl=%.docx))
PDFS  = $(subst $(DOCDIR),$(BLDDIR)pdf/,$(SRCS:%.rest=%.pdf))\
	$(subst $(DOCDIR),$(BLDDIR)pdf/,$(SRCSTPL:%.rest.stpl=%.pdf))
.PHONY: docx help Makefile docxdir pdfdir stpl index imgs
stpl: $(STPLTGTS)
%:%.stpl
	@cd $(DOCDIR) && stpl "$(<F)" "$(@F)"
imgs: $(PNGS)
$(DOCDIR)_images/%.png:$(DOCDIR)%.pyg
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)
$(DOCDIR)_images/%.png:$(DOCDIR)%.eps
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)
$(DOCDIR)_images/%.png:$(DOCDIR)%.tikz
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)
$(DOCDIR)_images/%.png:$(DOCDIR)%.svg
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)
$(DOCDIR)_images/%.png:$(DOCDIR)%.uml
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)
$(DOCDIR)_images/%.png:$(DOCDIR)%.dot
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)
docxdir: ${BLDDIR}docx
pdfdir: ${BLDDIR}pdf
MKDIR_P = mkdir -p
${BLDDIR}docx:
	@${MKDIR_P} ${BLDDIR}docx
${BLDDIR}pdf:
	@${MKDIR_P} ${BLDDIR}pdf
index:
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py
help:
	@$(SPHINXBLD) -M help "$(DOCDIR)" "$(BLDDIR)" $(SPHINXOPTS) $(O)
	@echo "  docx        to docx"
	@echo "  pdf         to pdf"
#http://www.sphinx-doc.org/en/stable/usage/builders/
html dirhtml singlehtml htmlhelp qthelp applehelp devhelp epub latex text man texinfo pickle json xml pseudoxml: Makefile index stpl imgs
	@$(SPHINXBLD) -M $@ "$(DOCDIR)" "$(BLDDIR)" $(SPHINXOPTS) $(O)
docx:  docxdir index stpl imgs $(DOCXS)
$(BLDDIR)docx/%.docx:$(DOCDIR)%.rest
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py "$(<F)" "$(DOCBACK)$@"
pdf: pdfdir index stpl imgs $(PDFS)
$(BLDDIR)pdf/%.pdf:$(DOCDIR)%.rest
	@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py "$(<F)" "$(DOCBACK)$@"

└ some.h
    /*
    #def gen_tst(lns,**kw):
    #  return [l.split('@')[1] for l in rlines(r'^\s*@',lns)]
    #def gen_tst
    #def gen_tstdoc(lns,**kw):
    #  return ['#) '+l.split('**')[1] for l in rlines(r'^/\*\*',lns)]
    #def gen_tstdoc

    @//generated from some.h
    @#include <assert.h>
    @#include "some.h"
    @int main()
    @{
    */

    /**Test add1()
    @assert(add1(1)==2);
    */
    int add1(int a)
    {
      return a+1;
    }

    /**Test add2()
    @assert(add2(1)==3);
    */
    int add2(int a)
    {
      return a+2;
    }

    /*
    @}
    */
```

# Output from make docx

When I run the makefile using make docx it throws the link 80 error missing separator.

```
➜  PostMortems rstdoc --stpl .
➜  PostMortems l
total 616
drwxr-xr-x  17 piercedevol  staff   544B Sep 11 14:13 .
drwxr-xr-x  19 piercedevol  staff   608B Sep 11 13:10 ..
-rw-r--r--@  1 piercedevol  staff    10K Sep 11 14:13 .DS_Store
-rw-r--r--   1 piercedevol  staff   3.4K Sep 11 14:13 Makefile
-rw-r--r--   1 piercedevol  staff   4.1K Sep 11 13:41 PostMortem-082020.rst
drwxr-xr-x   2 piercedevol  staff    64B Sep 11 14:13 build
-rw-r--r--   1 piercedevol  staff   3.1K Sep 11 14:13 conf.py
-rw-r--r--   1 piercedevol  staff   203K Sep 11 14:13 dcx.py
drwxr-xr-x  27 piercedevol  staff   864B Sep 11 14:13 doc
-rw-r--r--   1 piercedevol  staff    49B Sep 11 14:13 docutils.conf
-rw-r--r--   1 piercedevol  staff    15K Sep 11 14:13 reference.docx
-rw-r--r--   1 piercedevol  staff   9.0K Sep 11 14:13 reference.odt
-rw-r--r--   1 piercedevol  staff    21K Sep 11 14:13 reference.tex
-rw-r--r--   1 piercedevol  staff    39B Sep 11 14:13 waf
-rw-r--r--   1 piercedevol  staff   143B Sep 11 14:13 waf.bat
-rw-r--r--   1 piercedevol  staff   4.0K Sep 11 14:13 wafw.py
-rw-r--r--   1 piercedevol  staff   272B Sep 11 14:13 wscript
➜  PostMortems make docx
Makefile:80: *** missing separator.  Stop.
➜  PostMortems

```

# Additional info

A post I saw suggested that this output may be useful:

``` 
SPHINXOPTS  = -c .$
SPHINXBLD   = sphinx-build$
SPHINXPROJ  = sample$
DOCDIR      = doc/$
DOCBACK     = ../$
DCXFROMDOC  = ../$
BLDDIR      = build/doc/$
STPLS       = $(wildcard $(DOCDIR)*.stpl)$
STPLTGTS    = $(STPLS:%.stpl=%)$
SRCS        = $(filter-out $(DOCDIR)index.rest,$(wildcard $(DOCDIR)*.rest))$
SRCSTPL     = $(wildcard $(DOCDIR)*.rest.stpl)$
IMGS        = \$
^I$(wildcard $(DOCDIR)*.pyg)\$
^I$(wildcard $(DOCDIR)*.eps)\$
^I$(wildcard $(DOCDIR)*.tikz)\$
^I$(wildcard $(DOCDIR)*.svg)\$
^I$(wildcard $(DOCDIR)*.uml)\$
^I$(wildcard $(DOCDIR)*.dot)\$
^I$(wildcard $(DOCDIR)*.eps.stpl)\$
^I$(wildcard $(DOCDIR)*.tikz.stpl)\$
^I$(wildcard $(DOCDIR)*.svg.stpl)\$
^I$(wildcard $(DOCDIR)*.uml.stpl)\$
^I$(wildcard $(DOCDIR)*.dot.stpl)$
PNGS=$(subst $(DOCDIR),$(DOCDIR)_images/,\$
^I$(patsubst %.eps,%.png,\$
^I$(patsubst %.pyg,%.png,\$
^I$(patsubst %.tikz,%.png,\$
^I$(patsubst %.svg,%.png,\$
^I$(patsubst %.uml,%.png,\$
^I$(patsubst %.dot,%.png,\$
^I$(patsubst %.eps.stpl,%.png,\$
^I$(patsubst %.dot.stpl,%.png,\$
^I$(patsubst %.tikz.stpl,%.png,\$
^I$(patsubst %.svg.stpl,%.png,\$
^I$(patsubst %.uml.stpl,%.png,$(IMGS)))))))))))))$
DOCXS = $(subst $(DOCDIR),$(BLDDIR)docx/,$(SRCS:%.rest=%.docx))\$
^I$(subst $(DOCDIR),$(BLDDIR)docx/,$(SRCSTPL:%.rest.stpl=%.docx))$
PDFS  = $(subst $(DOCDIR),$(BLDDIR)pdf/,$(SRCS:%.rest=%.pdf))\$
^I$(subst $(DOCDIR),$(BLDDIR)pdf/,$(SRCSTPL:%.rest.stpl=%.pdf))$
.PHONY: docx help Makefile docxdir pdfdir stpl index imgs$
stpl: $(STPLTGTS)$
%:%.stpl$
^I@cd $(DOCDIR) && stpl "$(<F)" "$(@F)"$
imgs: $(PNGS)$
$(DOCDIR)_images/%.png:$(DOCDIR)%.pyg$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)$
$(DOCDIR)_images/%.png:$(DOCDIR)%.eps$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)$
$(DOCDIR)_images/%.png:$(DOCDIR)%.tikz$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)$
$(DOCDIR)_images/%.png:$(DOCDIR)%.svg$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)$
$(DOCDIR)_images/%.png:$(DOCDIR)%.uml$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)$
$(DOCDIR)_images/%.png:$(DOCDIR)%.dot$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py $(<F)$
docxdir: ${BLDDIR}docx$
pdfdir: ${BLDDIR}pdf$
MKDIR_P = mkdir -p$
${BLDDIR}docx:$
^I@${MKDIR_P} ${BLDDIR}docx$
${BLDDIR}pdf:$
^I@${MKDIR_P} ${BLDDIR}pdf$
index:$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py$
help:$
^I@$(SPHINXBLD) -M help "$(DOCDIR)" "$(BLDDIR)" $(SPHINXOPTS) $(O)$
^I@echo "  docx        to docx"$
^I@echo "  pdf         to pdf"$
#http://www.sphinx-doc.org/en/stable/usage/builders/$
html dirhtml singlehtml htmlhelp qthelp applehelp devhelp epub latex text man texinfo pickle json xml pseudoxml: Makefile index stpl imgs$
^I@$(SPHINXBLD) -M $@ "$(DOCDIR)" "$(BLDDIR)" $(SPHINXOPTS) $(O)$
docx:  docxdir index stpl imgs $(DOCXS)$
$(BLDDIR)docx/%.docx:$(DOCDIR)%.rest$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py "$(<F)" "$(DOCBACK)$@"$
pdf: pdfdir index stpl imgs $(PDFS)$
$(BLDDIR)pdf/%.pdf:$(DOCDIR)%.rest$
^I@cd $(DOCDIR) && python $(DCXFROMDOC)dcx.py "$(<F)" "$(DOCBACK)$@"$
$
‚M-^TM-^T some.h$
    /*$
    #def gen_tst(lns,**kw):$
    #  return [l.split('@')[1] for l in rlines(r'^\s*@',lns)]$
    #def gen_tst$
    #def gen_tstdoc(lns,**kw):$
    #  return ['#) '+l.split('**')[1] for l in rlines(r'^/\*\*',lns)]$
    #def gen_tstdoc$
$
    @//generated from some.h$
    @#include <assert.h>$
    @#include "some.h"$
    @int main()$
    @{$
    */$
$
    /**Test add1()$
    @assert(add1(1)==2);$
    */$
    int add1(int a)$
    {$
      return a+1;$
    }$
$
    /**Test add2()$
    @assert(add2(1)==3);$
    */$
    int add2(int a)$
    {$
      return a+2;$
    }$
$
    /*$
    @}$
    */$

```

# TLDR

I'm not sure why I'm unable to make the doc using make docx. I mean I'm sure it has to do with the error presented, but my skill level make it difficult. So I humbly ask for you help.

**Comment by rpuntaie** [comment]
This is a make error.
https://stackoverflow.com/questions/920413/make-error-missing-separator

Do you have this part with and after `some.h` in the Makefile?
If yes, how did it get there?
That should definitely not be there.

**Comment by rpuntaie** [comment]
OK, I found, how it got there.

`rstdoc --stpl .` has got a bug.

`rstdoc --stpl xyz`  is to generate/fill an example folder with `.stpl` files.
I think it does not cope with the `.` and expects a folder name instead.
`some.h` should have become a separate file,
which it does not with `.`.

I will look into this soon.
In the meantime don't use the `.`.
